### ISSUE TITLE
misunderstanding of ebm training

### ISSUE THREAD

**Issue by yyht** [salesforce/ebm_calibration_nlu#3]
Hi, this work is good one and i follow you work for my project, here are some misundstandings about you code:

logger.info('constructing datasets (splitting eval_dataset) for calibration...')
dataset_cal_dev1 = copy.deepcopy(eval_dataset)
dataset_cal_dev2 = copy.deepcopy(eval_dataset)
dataset_cal_tr = copy.deepcopy(train_dataset)
cal_num = int(len(eval_dataset) / 2)
dataset_cal_dev1.features = dataset_cal_dev1.features[:cal_num]
dataset_cal_dev2.features = dataset_cal_dev2.features[-cal_num:]
#dataset_cal_tr.features = dataset_cal_tr.features[-cal_num:]
logger.info('setting eval_dataset to dataset_cal_dev2...')
eval_dataset = dataset_cal_dev2

so, actually, you use eval_dataset for development and dataset_cal_dev1 for testing for all methods?
and the results from table1 and table2 is also not clear to me. test-set accuracy means test it on test-dataset splitted from normal glue deveploment dataset(dataset_cal_dev1 ?) since we can't not access to actual test dataset and estimate ece is impossible.

**Comment by cloudygoose** [comment]
Hi, it's the other way around.
The dataset_cal_dev2 (or eval_dataset) is used for actual evaluation (final results in table 1 and table 2)
I split the original dev set in half, precisely because as you said, we do not have the actual test dataset. So i'm using half of the original dev set as "test set", to compute ece and accuracy.
dataset_cal_dev1 is used for development.
